### ISSUE TITLE
Vague error with no clear way to track down where the bug is

### ISSUE THREAD

**Issue by SethTurin** [riot/compiler#107]
Hi

I've been trying to track down a bug, but the error is coming from riot, and it's giving me no useful information on how to find anything. I would narrow it down into something reproducible, but I can't do that either, because the app is several thousand lines of code and I have no indication where the problem is. Can you give me some idea of where I can look?

Here's there error:
```
VM190:3 Uncaught SyntaxError: Invalid or unexpected token
_create@ 
@ riot+compiler.js:636_tmpl 
@ riot+compiler.js:600updateExpression 
@ riot+compiler.js:1190each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266(anonymous function) 
@ riot+compiler.js:1567each 
@ riot+compiler.js:939updateEach 
@ riot+compiler.js:1540updateExpression 
@ riot+compiler.js:1187each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266updateExpression 
@ riot+compiler.js:1179each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266updateExpression 
@ riot+compiler.js:1179each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266updateDataIs 
@ riot+compiler.js:1124updateExpression 
@ riot+compiler.js:1213each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266(anonymous function) 
@ riot+compiler.js:1567each 
@ riot+compiler.js:939updateEach 
@ riot+compiler.js:1540updateExpression 
@ riot+compiler.js:1187each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266updateExpression 
@ riot+compiler.js:1179each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277update 
@ riot+compiler.js:1313updateExpression 
@ riot+compiler.js:1187each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266updateDataIs 
@ riot+compiler.js:1124updateExpression 
@ riot+compiler.js:1213each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266(anonymous function) 
@ riot+compiler.js:1567each 
@ riot+compiler.js:939updateEach 
@ riot+compiler.js:1540updateExpression 
@ riot+compiler.js:1187each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266updateExpression 
@ riot+compiler.js:1179each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266updateExpression 
@ riot+compiler.js:1179each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266updateExpression 
@ riot+compiler.js:1179each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266updateExpression 
@ riot+compiler.js:1179each 
@ riot+compiler.js:939updateAllExpressions 
@ riot+compiler.js:1277tagUpdate 
@ riot+compiler.js:2159tagMount 
@ riot+compiler.js:2266mountTo 
@ riot+compiler.js:2592pushTagsTo 
@ riot+compiler.js:1922each 
@ riot+compiler.js:939pushTagsTo 
@ riot+compiler.js:1927mount$2 
@ riot+compiler.js:1977(anonymous function) 
@ riot+compiler.js:3640done 
@ riot+compiler.js:3545compileScripts 
@ riot+compiler.js:3555compile 
@ riot+compiler.js:3629mount$$1 
@ riot+compiler.js:3640(anonymous function) 
@ index.html:20(anonymous function) 
@ lib.js?0.2623746091762764:1343(anonymous function) 
@ yaml.min.js:1f.onreadystatechange 
@ yaml.min.js:1

**Comment by GianlucaGuarini** [comment]
closed it's an invalid issue https://github.com/riot/riot/blob/master/.github/ISSUE_TEMPLATE.md#important

**Comment by SethTurin** [comment]
right, the problem is that I can't get more information about this issue because of the vagueness of the error. the only thing in that template that I can tell you is that I'm using the latest version, 3.7.4. any other information would require me to have some idea about which bit of code is causing the error, but riot doesn't tell me that.

If you can give me some clue as to how to narrow down the issue more, I'll happily provide more information.

**Comment by SethTurin** [comment]
to be clear, the issue that I'm reporting here is an issue with riot's error handling. there is likely a problem with my code or my setup, but it's impossible for me to find or narrow down due to the vagueness of the error being thrown by riot.

**Comment by SethTurin** [comment]
after hours of process of elimination, I managed to find the culprit. I had a line that looked like this:
flow-element(class!='{ "draggable-element": opts.editMode }')

for some reason this worked fine in chrome, but taking it to electron threw the error. I reversed the the order of the quotes to:
flow-element(class!="{ 'draggable-element': opts.editMode }")

and it started working fine. 

To re-emphasize the point of this thread, it would have saved me many hours of searching if riot had thrown an error that pointed to the faulty template. and at that point, if I was still confused, I would have been able to write a more informative ticket.
